lib/headerParsingLib.js

```diff
@@ -9,14 +8,11 @@ var querystring = require("querystring");
  */
 var getNextLink = function (headers) {
 	if (headers.link) {
-		var links = headers.link.split(",");
-		for (var elem in links) {
-			if (links[elem].indexOf("rel=\"next\"") !== -1) {
-				var linkRegex = /<(.*)>/;
-				var link = linkRegex.exec(links[elem])[1];
-				var parsedUrl = url.parse(link);
-				return querystring.parse(parsedUrl.query);
-			}
+		var parsedHeader = parse(headers.link);
+		if (parsedHeader.next) {
+			delete parsedHeader.next.rel;
+			delete parsedHeader.next.url;
```

Comment:
I'd recommend against ever really using the `delete` keyword (and many popular Javascript style guides agree). If we're really so specific about the returned object that we can't allow these fields to be present, we should be explicit in what we desire by creating a new object, selecting the keys we want.

```
return {
   sortKeyLT: parsedHeader.next.sortKeyLT
};
```

Being explicit protects us against any future changes to the `parse-link-header` library. If the maintainer decides to add a field to the `next` object, we won't have to come back to this code and `delete` it too.

As a rule of thumb in JS, I recommend never mutating objects if it's possible to avoid... really, that can be a good rule for development in general. Or at least, there is a growing trend that is pushing for immutability. It solves a lot of headaches as projects grow in complexity.